Unit Test(make test) FAIL: TestCacheBinary/ok_kubeadm

**The exact command to reproduce the issue**:

In macOS PC, when the following command exec, and we will get test fail.

```
make test
```

In `pkg/minikube/machine/pkg/minikube/machine/cache_binaries_test.go`,
Unit Test Case "ok kubeadm" of `TestCacheBinary` fails due to download error. 

```
{
	desc:         "ok kubeadm",
	version:      "v1.16.0",
	osName:       runtime.GOOS,
	archName:     runtime.GOARCH,
	binary:       "kubeadm",
	err:          false,
	minikubeHome: minikubeHome,
}
```

TestCacheBinary is the unit test which get binary(like kubeadm, kubelet, etc...) and put binary file to temp directory.

In the test case, minikube tries to download kubeadm according to the type of runtime.GOOS.

If runtime.GOOS is "darwin"(macOS), the kubeadm download URL is here:
`https://storage.googleapis.com/kubernetes-release/release/v1.16.0/bin/darwin/amd64/kubeadm`
However, the darwin binary is not prepared for kubeadm.
So, if we use macOS, this unit test always fails.

I'll fix this test to be sucessfull and I'll send PR soon.

**The full output of the command that failed**:<details>

Error Section

```
--- FAIL: TestCacheBinary (1.93s)
    --- FAIL: TestCacheBinary/ok_kubeadm (0.43s)
        cache_binaries_test.go:196: Got unexpected error Error downloading kubeadm v1.16.0: failed to download: failed to download to temp file: download failed: 1 error(s) occurred:

            * received invalid status code: 404 (expected 200)
127.0.0.1:64184
FAIL
```

make test full outout

```
./test.sh
= make lint =============================================================
ok
= go mod ================================================================
unused github.com/google/go-github/v25
ok
= boilerplate ===========================================================
ok
= schema_check ==========================================================
ok
= go test ===============================================================
ok  	k8s.io/minikube/cmd/minikube/cmd	0.113s	coverage: 18.5% of statements
ok  	k8s.io/minikube/cmd/minikube/cmd/config	0.076s	coverage: 20.5% of statements
ok  	k8s.io/minikube/pkg/drivers	0.066s	coverage: 13.4% of statements
ok  	k8s.io/minikube/pkg/drivers/hyperkit	0.038s	coverage: 13.8% of statements
ok  	k8s.io/minikube/pkg/minikube/assets	0.056s	coverage: 63.1% of statements
ok  	k8s.io/minikube/pkg/minikube/bootstrapper	1.778s	coverage: 73.3% of statements
ok  	k8s.io/minikube/pkg/minikube/bootstrapper/kubeadm	0.092s	coverage: 30.9% of statements
ok  	k8s.io/minikube/pkg/minikube/cluster	0.494s	coverage: 55.6% of statements
ok  	k8s.io/minikube/pkg/minikube/config	0.074s	coverage: 75.6% of statements
ok  	k8s.io/minikube/pkg/minikube/cruntime	0.020s	coverage: 64.0% of statements
ok  	k8s.io/minikube/pkg/minikube/extract	0.020s	coverage: 56.9% of statements
ok  	k8s.io/minikube/pkg/minikube/kubeconfig	0.060s	coverage: 75.9% of statements
ok  	k8s.io/minikube/pkg/minikube/logs	0.045s	coverage: 1.4% of statements
--- FAIL: TestCacheBinary (1.93s)
    --- FAIL: TestCacheBinary/ok_kubeadm (0.43s)
        cache_binaries_test.go:196: Got unexpected error Error downloading kubeadm v1.16.0: failed to download: failed to download to temp file: download failed: 1 error(s) occurred:

            * received invalid status code: 404 (expected 200)
127.0.0.1:64184
FAIL
coverage: 22.4% of statements
FAIL	k8s.io/minikube/pkg/minikube/machine	2.847s
ok  	k8s.io/minikube/pkg/minikube/notify	3.650s	coverage: 92.9% of statements
ok  	k8s.io/minikube/pkg/minikube/out	0.069s	coverage: 80.0% of statements
ok  	k8s.io/minikube/pkg/minikube/problem	0.019s	coverage: 100.0% of statements
ok  	k8s.io/minikube/pkg/minikube/proxy	0.031s	coverage: 100.0% of statements
ok  	k8s.io/minikube/pkg/minikube/registry	0.031s	coverage: 100.0% of statements
ok  	k8s.io/minikube/pkg/minikube/service	3.237s	coverage: 85.2% of statements
ok  	k8s.io/minikube/pkg/minikube/sshutil	0.552s	coverage: 100.0% of statements
ok  	k8s.io/minikube/pkg/minikube/storageclass	0.028s	coverage: 97.2% of statements
ok  	k8s.io/minikube/pkg/minikube/translate	0.017s	coverage: 11.9% of statements
ok  	k8s.io/minikube/pkg/minikube/tunnel	1.885s	coverage: 59.1% of statements
ok  	k8s.io/minikube/pkg/util	0.581s	coverage: 62.0% of statements
ok  	k8s.io/minikube/pkg/util/lock	0.010s	coverage: 59.1% of statements
ok  	k8s.io/minikube/pkg/util/retry	0.006s	coverage: 0.0% of statements
make: *** [test] Error 32
```

</details>

**The output of the `minikube logs` command**:<details>

nothing. because this case is download error(not minikube function).

</details>

**The operating system version**:

 * macOS 10.14.5
 * minikube(master branch source build at 2019/10/19)

--- a/pkg/minikube/machine/cache_binaries_test.go
+++ b/pkg/minikube/machine/cache_binaries_test.go
@@ -136,7 +136,7 @@ func TestCacheBinary(t *testing.T) {
 		{
 			desc:         "ok kubeadm",
 			version:      "v1.16.0",
-			osName:       runtime.GOOS,
+			osName:       "linux",
 			archName:     runtime.GOARCH,
 			binary:       "kubeadm",
 			err:          false,

